metricbeat/module/elasticsearch/ccr/ccr.go

```diff
@@ -107,3 +114,48 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) {
 		return
 	}
 }
+
+func (m *MetricSet) checkCCRAvailability(currentElasticsearchVersion string) (message string, err error) {
+	license, err := elasticsearch.GetLicense(m.HTTP, m.getResetURI())
+	if err != nil {
+		return "", errors.Wrap(err, "error determining Elasticsearch license")
+	}
+
+	licenseType, err := license.GetValue("type")
+	if err != nil {
+		return "", errors.Wrap(err, "error determining Elasticsearch license type")
+	}
+
+	currentLicense, ok := licenseType.(string)
+	if !ok {
+		return "", fmt.Errorf("error determining Elasticsearch license type")
+	}
+
+	isAvailable := currentLicense == "trial" || currentLicense == "platinum"
```

Comment:
Are you going to implement the generic license checking function in this PR? It could be basically the content of this function till here.
I guess we will have more cases where we have to check for the license.

Reply:
Yeah, I should do that. Thanks for catching. Will fix.